### PR TITLE
Fix nightly run CI

### PR DESCRIPTION
**Pull request by TianlongLiang**
Manually git clone and install dependencies instead of using git action on Ubuntu 14.04 runner

### CODE REVIEW

**Comment by TianlongLiang** [comment]
As mentioned in [this blog](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/), GitHub actions will default use node20, and we are using GH provided runner, we have no control over the external node version, temporarily replacing the GitHub action in `build_iwasm_linux_gcc4_8` with normal commands

**Comment by TianlongLiang** [comment]
@loganek @yamt what do you guys think?

File: .github/workflows/nightly_run.yml [hunk 1]
```diff
@@ -323,16 +323,13 @@ jobs:
           - make_options_run_mode: $FAST_JIT_BUILD_OPTIONS
             make_options_feature: "-DWAMR_BUILD_MEMORY64=1"
     steps:
-      - name: checkout
-        uses: actions/checkout@v3
```
Comment (yamt) [inline .github/workflows/nightly_run.yml:327]: can't you just bump to actions/checkout@v4?

File: .github/workflows/nightly_run.yml [hunk 2]
```diff
@@ -323,16 +323,13 @@ jobs:
...
-
       - name: Install dependencies
-        uses: nick-fields/retry@v2
```
Comment (yamt) [inline .github/workflows/nightly_run.yml:330]: the latest one seems to claim node 20 compat.
https://github.com/nick-fields/retry/releases/tag/v3.0.0

btw, is this retry necessary at all these days?

File: .github/workflows/nightly_run.yml (hunk 1 above)
Comment (TianlongLiang) [reply .github/workflows/nightly_run.yml:327]: The v4 will face the same issue, node 20 will require a more recent version of glibc and glibcxx than the Ubuntu 14.04 docker image provides

File: .github/workflows/nightly_run.yml (hunk 2 above)
Comment (TianlongLiang) [reply .github/workflows/nightly_run.yml:330]: I think it should be fine without it

File: .github/workflows/nightly_run.yml (hunk 1 above)
Comment (yamt) [reply .github/workflows/nightly_run.yml:327]: ah, i totally misread the PR description. sorry for noise!

**Review by yamt — APPROVED** [review]
looks reasonable to me

File: .github/workflows/nightly_run.yml (hunk 1 above)
Comment (TianlongLiang) [reply .github/workflows/nightly_run.yml:327]: No worries, I also made some modifications to my previous comments to make it clearer

**Review by wenyongh — APPROVED** [review]

**Comment by wenyongh** [comment]
I merged the PR since we had better let nightly-run CI run every day. @loganek please upload another PR to enhance it if needed, thanks.

**Pull request merged**